GCR/composite.py

```diff
@@ -17,22 +17,27 @@
 MATCHING_FORMAT = None
 MATCHING_ORDER = tuple()
 
-
-def _match(index_this, index_main, sorter=None):
-    if sorter is None:
+def _match(index_this, index_main, sorter=None, secondary=None, sort_descending=True):
+    if (sorter is None) and (secondary is None):
         sorter = np.argsort(index_this)
+    elif sorter is None:
+        sorter = np.argsort(secondary)
+        if sort_descending:
+            sorter = sorter[::-1]
+        sorter2 = np.argsort(index_this[sorter], kind='stable')
+        sorter = sorter[sorter2]
     s = np.searchsorted(index_this, index_main, sorter=sorter)
     s[s >= len(sorter)] = -1
     matching_idx = sorter[s]
-    not_matched_mask = (index_this[matching_idx] != index_main)
+    not_matched_mask = (index_this[matching_idx] != index_main) 
```

Comment:
```suggestion
    not_matched_mask = (index_this[matching_idx] != index_main)
```

removing trailing spaces